src/main/java/graphql/schema/FieldCoordinates.java

```diff
@@ -88,8 +108,28 @@ public static FieldCoordinates coordinates(String parentType, String fieldName)
      * @return the coordinates
      */
     public static FieldCoordinates systemCoordinates(String fieldName) {
-        assertTrue(fieldName.startsWith("__"), "Only __ system fields can be addressed without a parent type");
-        assertValidName(fieldName);
+        return systemCoordinates(fieldName, true);
+    }
+
+    /**
+     * The exception to the general rule is the system __xxxx Introspection fields which have no parent type and
+     * are able to be specified on any type.
+     *
+     * Note:  Whether or not validation is run depends on the {@code validate} boolean flag.  This method should only
+     *   be used in non-validation mode in cases where it's known that the values passed have been previously
+     *   validated.  That is, when those values are from objects such as {@code GraphQLFieldDefinition},
+     *   {@code GraphQLNamedOutputType}, etc.  The reason for this is to optimize the creation of the FieldCoordinates
+     *   object as validation can have a large impact on performance.
+     *
+     * @param fieldName the name of the system field which MUST start with __
+     *
+     * @return the coordinates
+     */
+    public static FieldCoordinates systemCoordinates(String fieldName, boolean validate) {
```

Comment:
As discussed - pleake make this private (and hence no need for the javadoc)

Reply:
>  The reason for this is to optimize the creation of the FieldCoordinates
     *   object as validation can have a large impact on performance.

Do you have any concrete evidence of the cost of these look ups?  I know regexes can be slow but I would love to see real perf data in how this lookup is used in real life.